oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/function/Suppliers.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.commons.function;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Supplier;
+
+/**
+ * Utility methods for {@link java.util.function.Supplier} handling.
+ */
+public class Suppliers {
+
+    private Suppliers() {
+    }
+
+    /**
+     * Transforms a {@link Supplier} based on a wrapper that evaluates
+     * the given {@code Supplier} at most one time.
+     * @param <T> return type
+     * @return Supplier based on the given Supplier
+     */
+    public static <@Nullable T> Supplier<T> memoize(final Supplier<@Nullable T> computeOnce) {
```

Comment:
Nullability annotations don't go on type parameters.

```suggestion
    public static <T> Supplier<@Nullable T> memoize(final Supplier<@Nullable T> computeOnce) {
```